src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,690 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml.XPath;
+
+using Mono.Cecil;
+
+namespace Mono.Linker.Dataflow
+{
+	class XmlFlowAnnotationSource : IFlowAnnotationSource
+	{
+		readonly Dictionary<MethodDefinition, AnnotatedMethod> _methods = new Dictionary<MethodDefinition, AnnotatedMethod> ();
+		readonly Dictionary<PropertyDefinition, DynamicallyAccessedMemberTypes> _properties = new Dictionary<PropertyDefinition, DynamicallyAccessedMemberTypes> ();
+		readonly Dictionary<FieldDefinition, DynamicallyAccessedMemberTypes> _fields = new Dictionary<FieldDefinition, DynamicallyAccessedMemberTypes> ();
+
+		static readonly string _signature = "signature";
```

Comment:
I don't generally find introducing constants for things that only occur once very useful. Could you inline them, or at least make them `const string` instead of `static readonly string`?